### PR TITLE
(WSR 8) <RichTextInputArea/> - error & errorMessage -> status & statusMessage

### PR DESCRIPTION
**Pull request by EgoziE**
- Props:
  - changed `error` to `status`
  - changed `errorMessage` to `statusMessage`
- Testkit:
  - removed `hasError`
  - removed `getErrorMessage`
  - added `hasStatus`
  - added `getStatus`
  - added `hasStatusMessage`
  - added `getStatusMessage`

### CODE REVIEW

File: src/RichTextInputArea/RichTextInputArea.js [hunk 1]
```diff
@@ -82,7 +79,8 @@ class RichTextInputArea extends React.PureComponent {
           styles.root,
           !isEditorEmpty && styles.hidePlaceholder,
           disabled && styles.disabled,
-          hasError && styles.error,
+          !disabled && status === 'error' && styles.hasError,
+          !disabled && status === 'warning' && styles.hasWarning,
```
Comment (simast) [inline src/RichTextInputArea/RichTextInputArea.js]: Lets maybe use built-in support to toggling class names:
```js
className={classNames(styles.root, {
  [styles.hidePlaceholder]: !isEditorEmpty,
  [styles.disabled]: !!disabled,
  [styles.hasError]: !disabled && status === 'error',
  [styles.hasWarning]: !disabled && status === 'warning',
})}
```

File: src/RichTextInputArea/index.d.ts [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+
 export interface RichTextInputAreaProps {
   dataHook?: string;
   initialValue?: string;
   placeholder?: string;
   disabled?: boolean;
-  status?: 'error';
-  statusMessage?: string;
+  status?: RichTextInputAreaStatus;
+  statusMessage?: React.ReactNode;
```
Comment (simast) [inline src/RichTextInputArea/index.d.ts]: not directly related to this PR, but if you will be making any further changes, could you update InputArea types to be a `React.ReactNode` as well?
https://github.com/wix/wix-style-react/blob/025c48edf9c412f8d906ed041e7975a6ee310c10/src/InputArea/index.d.ts#L48

File: src/RichTextInputArea/test/RichTextInputArea.spec.js [hunk 3]
```diff
@@ -4,11 +4,16 @@ import eventually from 'wix-eventually';
 
 import RichTextInputArea from '../RichTextInputArea';
 import richTextInputAreaPrivateDriverFactory from '../RichTextInputArea.private.uni.driver';
-import toolbarButtonStyles from '../RichTextToolbarButton.scss';
+import toolbarButtonStyles from '../Toolbar/RichTextToolbarButton.scss';
```
Comment (simast) [inline src/RichTextInputArea/test/RichTextInputArea.spec.js]: 🤔how come the tests passed before this change?

Comment (EgoziE) [reply src/RichTextInputArea/test/RichTextInputArea.spec.js]: I have no idea 😳
It seems this test can be re-written differently (but not in this PR)

File: src/RichTextInputArea/index.d.ts (hunk 2 above)
Comment (EgoziE) [reply src/RichTextInputArea/index.d.ts]: sure

**Pull request merged**